modules/apps/frontend-data-set/frontend-data-set-admin-web/src/main/java/com/liferay/frontend/data/set/admin/web/internal/portlet/action/ImportSystemDataSetMVCResourceCommand.java

```diff
@@ -176,92 +284,97 @@ private void _addFDSCreationMenuObjectEntries(
 						"confirmationMessageType",
 						() -> _getOptionalValue(
 							data.get("confirmationMessageType"))
+					).put(
+						"errorMessage_i18n",
+						() -> _getLocalizeableValue(
+							defaultLanguageId,
+							_getOptionalValue(data.get("errorMessage")))
+					).put(
+						"icon",
+						() -> _getOptionalValue(
+							fdsActionDropdownItem.get("icon"))
+					).put(
+						"label_i18n",
+						() -> _getLocalizeableValue(
+							defaultLanguageId,
+							_getOptionalValue(
+								fdsActionDropdownItem.get("label")))
+					).put(
+						"method", () -> _getOptionalValue(data.get("method"))
 					).put(
 						"modalSize",
 						() -> _getOptionalValue(data.get("modalSize"))
 					).put(
 						"permissionKey",
 						() -> _getOptionalValue(data.get("permissionKey"))
+					).put(
+						"r_dataSetToDataSetActions_l_dataSetId", dataSetId
+					).put(
+						"requestBody",
+						() -> _getOptionalValue(data.get("requestBody"))
+					).put(
+						"successMessage_i18n",
+						() -> _getLocalizeableValue(
+							defaultLanguageId,
+							_getOptionalValue(data.get("successMessage")))
+					).put(
+						"target",
+						String.valueOf(fdsActionDropdownItem.get("target"))
 					).put(
 						"title_i18n",
 						() -> _getLocalizeableValue(
 							defaultLanguageId,
 							_getOptionalValue(data.get("title")))
-					).build());
+					).put(
+						"type", "item"
+					).put(
+						"url",
+						() -> _getOptionalValue(
+							fdsActionDropdownItem.get("href"))
+					).build(),
+					new ServiceContext());
 			}
-
-			_objectEntryService.addObjectEntry(
-				0, objectDefinitionId, objectEntryValues, new ServiceContext());
 		}
-	}
-
-	private void _addFDSItemsActionsObjectEntries(
-			long dataSetId, String defaultLanguageId,
-			FDSItemsActions fdsItemsActions,
-			HttpServletRequest httpServletRequest, long objectDefinitionId)
-		throws Exception {
-
-		List<FDSActionDropdownItem> fdsActionDropdownItems =
-			fdsItemsActions.getFDSActionDropdownItems(httpServletRequest);
+		else if (fdsEntryItemImportPolicy ==
+					FDSEntryItemImportPolicy.ITEM_PROXY) {
 
-		for (FDSActionDropdownItem fdsActionDropdownItem :
-				fdsActionDropdownItems) {
+			for (FDSActionDropdownItem fdsActionDropdownItem :
+					fdsActionDropdownItems) {
 
-			Map<String, Object> data =
-				(Map<String, Object>)fdsActionDropdownItem.get("data");
+				Map<String, Object> data =
+					(Map<String, Object>)fdsActionDropdownItem.get("data");
 
+				_objectEntryService.addOrUpdateObjectEntry(
+					String.valueOf(data.get("id")), 0, objectDefinitionId,
```

Comment:
Even if we know we have `id` it can come as a `null` because `HashMap` allows them as values. So I'd say we can apply the same logic we implemented for `ITEM_PROXY` creation menu entries import